pkg/executor/aggregate/agg_hash_partial_worker.go

```diff
@@ -67,43 +89,140 @@ func (w *HashAggPartialWorker) getChildInput() bool {
 	return true
 }
 
+func (w *HashAggPartialWorker) fetchChunkAndProcess(ctx sessionctx.Context, hasError *bool, needShuffle *bool, enableIntest bool) bool {
+	w.spillHelper.syncLock.RLock()
+	defer w.spillHelper.syncLock.RUnlock()
+
+	if w.spillHelper.isInSpilling() {
+		// Repeatedly release lock so that it could stop processing data when
+		// spill is in execution and spill action could have more chance to get write lock.
```

Comment:
Should add a sleep here to avoid repeatedly getting and releasing the lock?

Reply:
> Should add a sleep here to avoid repeatedly getting and releasing the lock?

okk